src/lib/re.js

```diff
@@ -220,89 +218,227 @@ var $builtinmodule = function (name) {
         }
-        var mob = Sk.misceval.callsim(mod.MatchObject, lst, pattern, string);
+        mob = Sk.misceval.callsim(mod.MatchObject, lst, pattern, string);
         return mob;
-    });
+    };
+
+    _search.co_varnames = ["pattern", "string", "flags"];
+    _search.co_numargs = 3;
+    _search.$defaults = [ new Sk.builtin.int_(0) ];
 
-    mod.match = new Sk.builtin.func(function (pattern, string, flags) {
-        Sk.builtin.pyCheckArgs('match', arguments, 2, 3);
+    mod.search = new Sk.builtin.func(_search);
+
+    _match = function (pattern, string, flags) {
+        var mob, res;
+        Sk.builtin.pyCheckArgs("match", arguments, 2, 3);
         if (!Sk.builtin.checkString(pattern)) {
             throw new Sk.builtin.TypeError("pattern must be a string");
         }
-        ;
         if (!Sk.builtin.checkString(string)) {
             throw new Sk.builtin.TypeError("string must be a string");
         }
-        ;
         if (flags === undefined) {
             flags = 0;
         }
-        ;
         if (!Sk.builtin.checkNumber(flags)) {
             throw new Sk.builtin.TypeError("flags must be a number");
         }
-        ;
-        var res = "/^" + pattern.v.replace(/\//g, "\\/") + "/";
+        pat = Sk.ffi.remapToJs(pattern);
+        res = "/^" + pat.replace(/\//g, "\\/") + "/";
         lst = mod._findre(res, string);
-        if (lst.v.length < 1) {
+        if (Sk.ffi.remapToJs(lst).length < 1) {
             return Sk.builtin.none.none$;
         }
-        var mob = Sk.misceval.callsim(mod.MatchObject, lst, pattern, string);
+        mob = Sk.misceval.callsim(mod.MatchObject, lst, pattern, string);
         return mob;
+    };
+
+    _match.co_varnames = ["pattern", "string", "flags"];
+    _match.co_numargs = 3;
+    _match.$defaults = [ new Sk.builtin.int_(0) ];
+
+    mod.match = new Sk.builtin.func(_match);
+
+    regexobj = function ($gbl, $loc) {
+        var _slice, _re_search, _re_match, _re_split, _re_findall, _repr;
+
+        $loc.__init__ = new Sk.builtin.func(function (self, pattern, flags) {
+            self.re = pattern;
+            if (flags === undefined) {
+                self.flags = 0;
+            }
+            else {
+                self.flags = flags;
+            }
+        });
+
+        _repr = new Sk.builtin.func( function (self) {
+            var ret = "re.compile('" + Sk.ffi.remapToPy(self.re) + "')";
+            return Sk.ffi.remapToPy(ret.substring(0,212));
+        });
+
+        $loc.__str__ = _repr;
+
+        $loc.__repr__ = _repr;
+
+        // Given a string, start, and end position, return sliced string
+        _slice = function(string, pos, endpos) {
+            // Per docs, ^ should match index after newlines.
+            // this matches the first
+            var str = Sk.ffi.remapToJs(string);
+            var start = pos == undefined ? 0 : Sk.ffi.remapToJs(pos);
+            var end = endpos == undefined ? str.length : Sk.ffi.remapToJs(endpos);
```

Comment:
@eah13 these two `undefined` checks were added ti fix the bug @bnmnetp found

Reply:
Ah yes.  Thanks!